crates/miden-objects/src/address/routing_parameters.rs

```diff
@@ -173,36 +200,23 @@ impl RoutingParameters {
     ) -> Result<Self, AddressError> {
         let mut interface = None;
         let mut note_tag_len = None;
+        let mut encryption_key = None;
 
         while let Some(key) = byte_iter.next() {
             match key {
                 RECEIVER_PROFILE_PARAM_KEY => {
-                    if byte_iter.len() < 2 {
-                        return Err(AddressError::decode_error(
-                            "expected two bytes to decode receiver profile",
-                        ));
-                    };
-
-                    let byte0 = byte_iter.next().expect("byte0 should exist");
-                    let byte1 = byte_iter.next().expect("byte1 should exist");
-                    let receiver_profile = u16::from_be_bytes([byte0, byte1]);
-
-                    let tag_len = (receiver_profile >> 11) as u8;
-                    note_tag_len = if tag_len == ABSENT_NOTE_TAG_LEN {
-                        None
-                    } else {
-                        Some(tag_len)
-                    };
-
-                    let addr_interface = receiver_profile & 0b0000_0111_1111_1111;
-                    let addr_interface =
-                        AddressInterface::try_from(addr_interface).map_err(|err| {
-                            AddressError::decode_error_with_source(
-                                "failed to decode address interface",
-                                err,
-                            )
-                        })?;
-                    interface = Some(addr_interface);
+                    if interface.is_some() {
+                        return Err(AddressError::decode_error("duplicate receiver profile tag"));
```

Comment:
```suggestion
                        return Err(AddressError::decode_error("duplicate receiver profile routing parameter"));
```

Nit: I'd call this routing parameter instead of tag to stay consistent with terminology.